jobs/gcp-discovery/src/main/java/com/tmobile/pacbot/gcp/inventory/file/AssetFileGenerator.java

```diff
@@ -196,46 +196,46 @@ public void generateFiles(List<ProjectVH> projects, String filePath) {
 					e.printStackTrace();
 				}
 			});
-			executor.execute(() -> {
-				if (!(isTypeInScope("clouddns"))) {
-					return;
-				}
-				try {
-					FileManager.generateCloudDnsFiles(cloudDNSInventoryCollector.fetchCloudDnsInventory(project));
-				} catch (Exception e) {
-					e.printStackTrace();
-				}
-			});
-			executor.execute(() -> {
-				if (!(isTypeInScope("networks"))) {
-					return;
-				}
-				try {
-					FileManager.generateNetworksFiles(networkInventoryCollector.fetchNetworkInventory(project));
-				} catch (Exception e) {
-					e.printStackTrace();
-				}
-			});
-			executor.execute(() -> {
-				if (!(isTypeInScope("project"))) {
-					return;
-				}
-				try {
-					FileManager.generateProjectFiles(projectInventoryCollector.fetchProjectMetadataMetadata(project));
-				} catch (Exception e) {
-					e.printStackTrace();
-				}
-			});
-			executor.execute(() -> {
-				if (!(isTypeInScope("serviceaccounts"))) {
-					return;
-				}
-				try {
-					FileManager.generateServiceAccountFiles(serviceAccountInventoryCollector.fetchServiceAccountDetails(project));
-				} catch (Exception e) {
-					e.printStackTrace();
-				}
-			});
+//			executor.execute(() -> {
+//				if (!(isTypeInScope("clouddns"))) {
+//					return;
+//				}
+//				try {
+//					FileManager.generateCloudDnsFiles(cloudDNSInventoryCollector.fetchCloudDnsInventory(project));
+//				} catch (Exception e) {
+//					e.printStackTrace();
+//				}
+//			});
+//			executor.execute(() -> {
+//				if (!(isTypeInScope("networks"))) {
+//					return;
+//				}
+//				try {
+//					FileManager.generateNetworksFiles(networkInventoryCollector.fetchNetworkInventory(project));
+//				} catch (Exception e) {
+//					e.printStackTrace();
+//				}
+//			});
+//			executor.execute(() -> {
+//				if (!(isTypeInScope("project"))) {
+//					return;
+//				}
+//				try {
+//					FileManager.generateProjectFiles(projectInventoryCollector.fetchProjectMetadataMetadata(project));
+//				} catch (Exception e) {
+//					e.printStackTrace();
+//				}
+//			});
+//			executor.execute(() -> {
+//				if (!(isTypeInScope("serviceaccounts"))) {
+//					return;
+//				}
+//				try {
+//					FileManager.generateServiceAccountFiles(serviceAccountInventoryCollector.fetchServiceAccountDetails(project));
+//				} catch (Exception e) {
+//					e.printStackTrace();
+//				}
+//			});
 
 			executor.shutdown();
 
```

Comment:
```while (!executor.isTerminated()) {
			}```
			these kind of checks can hog cpu. Please add a sleep inside even if for 500ms.

Reply:
We will add this. thanks for the comment.